exporter/collector/integrationtest/cmd/recordfixtures/main.go

```diff
@@ -54,12 +54,23 @@ func main() {
 	startTime := endTime.Add(-time.Second)
 	t := &FakeTesting{}
 
-	recordLogs(ctx, t, endTime)
-	recordMetrics(ctx, t, startTime, endTime)
-	recordTraces(ctx, t, startTime, endTime)
+	// track the output fixtures to catch tests overwriting another test's fixtures
```

Comment:
Could the fixture path be generated deterministically from the test name instead (so that collisions would be impossible)?

If we need to maintain a mapping, perhaps it would be more readable to define a type for the fixture tracker. One reason is that I usually expect functions to modify their receiver, but not to modify parameters that are passed in.

```go
type fixtureTracker map[string]string
func (ft fixtureTracker) register(test testcases.TestCase) error {...}
func (ft fixtureTracker) recordTraces(ctx context.Context, t *FakeTesting, startTime, endTime time.Time) {...}
```
```

Reply:
sgtm, makes it easier to add other checks too if we need to